cypress/support/step_definitions/fireboltCalls.js

```diff
@@ -527,15 +527,26 @@ Given(
 );
 
 /**
- * @function User triggers event with value '{}}'
- * @description sending message to platform to make post call to set values.
- * @param {String} key - Name of event to be called.
+ * @function User triggers event with value as '(.+)'
+ * @description sending message to platform to make post call to set event values.
+ * @param {String} key - key name of the event data
  * @example
- * And User triggers event with value as ' DEVICE_ONHDCPCHANGED_EVENTS'
+ * And User triggers event with value as 'onNetworkChanged events with wifi connected'
  */
 Given(/User triggers event with value as '(.+)'/, (key) => {
-  fireLog.info(CONSTANTS.STEP_DEFINITION_NEEDS_TO_IMPLEMENT).then(() => {
-    throw new Error(CONSTANTS.STEP_IMPLEMENTATION_MISSING);
+  cy.fireboltDataParser(key).then((parsedData) => {
+    const value = parsedData[0].params;
```

Comment:
Curious about the data being an array and statically taking the first value. Can you explain this?

Reply:
This is because current fireboltDataParser implementation returns data as an array and not object. Also in our case since we are setting event value for a given event passed in key, the array will only have 1 data object and hence the code

The basic intention of the ticket was to unblock Snehal's ticket FIRECERT-2171 which needs this glue support. All the json objects are added in fireboltCalls as part of this ticket (https://github.com/rdkcentral/firebolt-certification-suite/pull/134/files), and here I am trying to parse through that data and set values. I remember giving a try to move a value to fireboltMocks and use getFireboltData() instead, but I was facing errors in that case. So the decision was to go ahead with fireboltDataParser as the latter is outside the scope of this ticket and might need additional work if to be taken as a seperate ticket

Reply:
Understood, thank you for elaborating.